src/components/spreadsheet/custom-columns/use-custom-column.ts

```diff
@@ -0,0 +1,136 @@
+                },
+                evaluate: function () {
+                    throw new Error('Function evaluate is disabled');
+                },
+                parse: function () {
+                    throw new Error('Function parse is disabled');
+                },
+                simplify: function () {
+                    throw new Error('Function simplify is disabled');
+                },
+                derivative: function () {
+                    throw new Error('Function derivative is disabled');
+                },
+                equal: function (a: any, b: any) {
+                    // == instead of === to be able to compare strings to numbers
+                    return a === b;
+                },
+            },
+            { override: true }
+        );
+        return { limitedEvaluate };
+    }, []);
+
+    // Main function to sort columns by dependencies and calculate values
+    const sortedColumnDefinitions = useMemo(() => {
+        const graph = createDependencyGraph(customColumnsDefinitions);
+        const sortedColumns = topologicalSort(graph);
+
+        return sortedColumns.map((name) =>
+            customColumnsDefinitions.find((col) => col.name === name)
+        ) as ColumnWithFormula[];
+    }, [customColumnsDefinitions]);
+
+    const calcAllColumnValues = useCallback(
+        (lineData: Record<string, unknown>): Map<string, unknown> => {
+            const customColumnsValues = new Map<string, unknown>();
+            const scope: Record<string, unknown> = {};
+
+            // Add line data to the scope
+            Object.entries(lineData).forEach(([key, value]) => {
+                scope[`${key}`] = typeof value === 'number' ? bignumber(value) : value;
+            });
+
+            sortedColumnDefinitions.forEach((column) => {
+                if (!column) {
+                    return;
+                }
+                try {
+                    // Evaluate the formula and update the Map and scope
+                    const result = math.limitedEvaluate(column.formula, {
+                        ...scope,
+                        ...Object.fromEntries(customColumnsValues),
+                    });
+
+                    customColumnsValues.set(column.name, result);
+                    scope[column.name] = result; // Add calculated result to scope for future columns
+                } catch (error: any) {
+                    console.error(`Error evaluating formula for column ${column.name}: ${error.message}`);
+                }
+            });
+
+            return customColumnsValues;
+        },
+        [math, sortedColumnDefinitions]
+    );
+
+    const createCustomColumn = useCallback(() => {
+        return customColumnsDefinitions.map((colWithFormula: ColumnWithFormula) => {
+            return makeAgGridCustomHeaderColumn({
+                headerName: colWithFormula.name,
+                field: colWithFormula.name,
+                numeric: true,
```

Comment:
I think you can remove numeric, filterParams, cellDataType, cellRenderer for now because it can be several data types. We will improve that later on :)

Reply:
For cellRender, the same remark as above